### PR TITLE
jest config, mainPage and cardItem tests

### PR DESCRIPTION
**Pull request by XSpiritBalanceX**
application component testing

### CODE REVIEW

File: src/pages/MyPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        headers: {
+          Authorization: "Bearer null",
+          "Content-type": "application/json",
+        },
+        method: "POST",
+      }
+    );
+    expect(fetch).toHaveBeenCalledTimes(1);
+  });
+});
```
Comment (XSpiritBalanceX) [inline src/pages/MyPage.test.tsx:28]: how to change the isLogin state to test the rest of the page

File: src/components/LoginForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    render(<LoginForm />, { wrapper: BrowserRouter });
+    const emailField: HTMLInputElement = screen.getByLabelText("email");
+    const passwordField: HTMLInputElement = screen.getByLabelText("password");
+
+    fireEvent.change(emailField, { target: { value: testValues.email } });
+    expect(emailField.value).toBe(testValues.email);
+
+    fireEvent.change(passwordField, { target: { value: testValues.password } });
+    expect(passwordField.value).toBe(testValues.password);
+  });
```
Comment (XSpiritBalanceX) [inline src/components/LoginForm.test.tsx:49]: how to test onSubmitLogin function after click on submit button

File: src/components/RegistrationForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    expect(passwordField.value).toBe(testValues.password);
+  });
+
+  it("should check fields of form for password match", () => {
+    const { passwordField, repeatField } = setup();
+
+    fireEvent.change(passwordField, { target: { value: testValues.password } });
+    fireEvent.change(repeatField, { target: { value: testValues.password } });
+    expect(repeatField.value).toEqual(passwordField.value);
+  });
```
Comment (XSpiritBalanceX) [inline src/components/RegistrationForm.test.tsx:72]: how to test onSubmitRegistration function after click on submit button

File: src/pages/MyPage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+jest.mock("../hooks/useTypedSelector", () => ({
+  useTypedSelector: () => {
+    return true;
+  },
+}));
+
+jest.mock("react-redux", () => ({
+  useDispatch: jest.fn(),
+}));
```
Comment (XSpiritBalanceX) [inline src/pages/MyPage.test.tsx]: how to test the dispatch function

File: src/components/RegistrationForm.test.tsx (hunk 3 above)
Comment (OlegPanasyuk) [reply src/components/RegistrationForm.test.tsx:72]: Mock useForm hook. Then you will get ability to provide the callback function for sumbit handler

File: src/components/LoginForm.test.tsx (hunk 2 above)
Comment (OlegPanasyuk) [reply src/components/LoginForm.test.tsx:49]: Mock useForm hook. Then you will get ability to provide the callback function for sumbit handler

File: src/pages/MyPage.test.tsx (hunk 1 above)
Comment (OlegPanasyuk) [reply src/pages/MyPage.test.tsx:28]: Mock fetch with resolved values

File: package.json [hunk 5]
```diff
@@ -54,9 +66,10 @@
     "bootstrap-icons": "^1.10.3",
     "classnames": "^2.3.2",
     "concurrently": "^7.6.0",
+    "history": "^5.3.0",
     "ky": "^0.33.2",
```
Comment (OlegPanasyuk) [inline package.json]: If we are not use the ky package let's remove it from deps

File: src/components/CardItem.test.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  description:
+    "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...",
+  dateofnewseason: "закрыт",
+};
+
+describe("Card item test:", () => {
+  it("should render info about data of new seasons", async () => {
+    render(<CardItem infoSeries={fakeInfoSeries} />);
+    let dataSeason;
+    if (fakeInfoSeries.dateofnewseason !== "закрыт") {
```
Comment (OlegPanasyuk) [inline src/components/CardItem.test.tsx]: There are should be two test cases. Any 'if' constructions in tests, except throw the error.

File: src/components/RegistrationForm.test.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    expect(nameField.value).toBe(testValues.name);
+
+    fireEvent.change(emailField, { target: { value: testValues.email } });
+    expect(emailField.value).toBe(testValues.email);
+
+    fireEvent.change(passwordField, { target: { value: testValues.password } });
+    expect(passwordField.value).toBe(testValues.password);
+  });
+
+  it("should check fields of form for password match", () => {
```
Comment (OlegPanasyuk) [inline src/components/RegistrationForm.test.tsx]: Probably it's better to check if there are any validation error messages

File: src/components/RegistrationForm.test.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    expect(screen.getByTestId("form")).toBeInTheDocument();
+  });
+
+  it("should get the url with the path= /login if user click on link", () => {
+    render(<RegistrationForm />, { wrapper: BrowserRouter });
+    fireEvent.click(screen.getByText("Авторизоваться"));
+    expect(global.window.location.href).toContain("http://localhost/login");
+  });
+
+  it("should check fields of form when user tries to fill them in", () => {
```
Comment (OlegPanasyuk) [inline src/components/RegistrationForm.test.tsx]: I'm not sure that the sense of the statement is relevant for the asserts below it

File: src/components/fetchWrapper.ts [hunk 9]
```diff
@@ -21,13 +20,19 @@ export const fetchWrapper = {
...
+    const response = await fetch(APIRouters.registration, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
     });
     if (response.status !== 200) {
       toast.error(response.statusText);
```
Comment (OlegPanasyuk) [inline src/components/fetchWrapper.ts]: Can we move these toasts to the error Boundary component?

File: src/hooks/useAuthFetch.test.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+}));
+
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    json: () => Promise.resolve([]),
+  })
+) as jest.Mock;
+
+const token =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InVzZXJAdXNlci5jb20iLCJwYXNzd29yZCI6IjEyMzQ1NiJ9.lvxQl6QQAsTgaaQi59yDHK43aWc7JwTlXKtQgJpiI6g";
```
Comment (OlegPanasyuk) [inline src/hooks/useAuthFetch.test.ts]: It shouldn't be real data. Let's use the 'token.string'

**Review by OlegPanasyuk — APPROVED** [review]

**Pull request merged**